src/middlewared/middlewared/plugins/smb.py

```diff
@@ -248,6 +251,7 @@ class Config:
         verrors = ValidationErrors()
         path = data.get('path')
         default_perms = data.pop('default_permissions', False)
+        data['vuid'] = await self.generate_vuid(data['timemachine'], data['vuid'])
```

Comment:
data['timemachine'] is not guaranteed to exist at this point, `timemachine` may not be provided for update.

We should move this after `new`:
new['vuid'] = await self.generate_vuid(new['timemachine'], new['vuid'])